stock_by_warehouse/models/product_product.py

```diff
@@ -0,0 +1,187 @@
+        self.warehouses_stock_recompute = True
+
+    def _product_available_not_res_hook(self, quants):
+        """Hook used to introduce possible variations"""
+        return False
+
+    def _prepare_domain_available_not_reserved(self):
+        domain_quant = [
+            ('product_id', 'in', self.ids)
+        ]
+        domain_quant_locations = self._get_domain_locations()[0]
+        domain_quant.extend(domain_quant_locations)
+        return domain_quant
+
+    @api.depends_context('warehouse', 'force_company')
+    def _compute_qty_available_not_reserved(self):
+        """Method to set the quantity available that is not reserved of a serie of products.
+        """
+        domain_quant = self._prepare_domain_available_not_reserved()
+        quants = self.env['stock.quant'].with_context(lang=False).read_group(
+            domain_quant, fields=['product_id', 'quantity', 'reserved_quantity'], groupby="product_id")
+
+        res = {}
+        for quant in quants:
+            quantity = quant.get('quantity') - quant.get('reserved_quantity')
+            res[quant.get('product_id')[0]] = max(quantity, 0.0)
+
+        for prod in self:
+            prod.qty_available_not_res = res.get(prod.id, 0.0)
+
+    def _compute_get_quantity_warehouses_json(self):
+        # get original from onchange
+        self_origin = self._origin if hasattr(self, '_origin') else self
+        info = {'title': _('Stock by Warehouse'), 'content': [],
+                'warehouse': self_origin.qty_available_not_res}
+        if not self_origin.exists():
+            return json.dumps(info)
+        self_origin.ensure_one()
+
+        # Just in case it's asked from other place different than product
+        # itself, we enable this context management
+        warehouse_id = self._context.get('warehouse_id')
+
+        for warehouse in self.env['stock.warehouse'].search([]):
+            product = self_origin.with_context(
+                warehouse=warehouse.id, location=False, force_company=warehouse.company_id.id)
+            if warehouse_id and warehouse_id.id == warehouse.id:
+                info['warehouse'] = product.qty_available_not_res
+            info['content'].append({
+                'warehouse': warehouse.name,
+                'warehouse_short': warehouse.code,
+                'product': product.id,
+                'available_not_res': product.qty_available_not_res,
+                'available': product.qty_available,
+                'virtual': product.virtual_available,
+                'incoming': product.incoming_qty,
+                'outgoing': product.outgoing_qty,
+                'saleable':
+                product.qty_available - product.outgoing_qty
+            })
+        return json.dumps(info)
+
+    @api.depends_context(
+        'lot_id', 'owner_id', 'package_id', 'from_date', 'to_date',
+        'company_owned', 'force_company', 'warehouse',
+    )
+    def _compute_quantities(self):
```

Comment:
@hugho-ad 

This inheritance is no longer necessary, the missing field "warehouse" was fixed in this commit  https://github.com/odoo/odoo/commit/d6bd78e578104f43318fbe3c11f325063e3e5042  and V14.0 it is also fixed https://github.com/odoo/odoo/blob/14.0/addons/stock/models/product.py#L100 

Regards !